### ISSUE TITLE
GB - JWT not working (resolved, my user error!)

### ISSUE THREAD

**Issue by tdoherty89** [Sholofly/lghorizon#152]
UPDATE:  Thanks to users ColinRobbins and mpsamuel for pointing out that I had used the wrong JWT Token.

Hiya,

Thanks for all you do, Sholofly. I completely understand you do this off your own back, and free of charge, so I don't expect a fix. I only wanted to highlight that this doesn't seem to be working either for someone to point out what I am doing wrong, or for your awareness.

Background: Previous iterations of this integration worked for me (i.e. Arris DCX960) until the GB issues arose maybe last year. I am based in the UK and am a Virgin Media customer. I have been trying to install this again for a couple of weeks since the JWT solution was recently made available in the latest updates.

Further context: I am following the latest instructions for the UK. I am using Firefox and JWT Debugger to get the JWT token. I am copying the whole thing. My 360 box standby power consumption is set to "Fast start", which is the least efficient option of the three options available. I have installed the latest updates, and am at version v0.7.4.

Error: When trying to install, i choose Virgin (GB), I enter my email address as the username and click enter. It takes me to the next dialogue box that asks me for the JWT, that i paste. I then get the error message, "Can't Connect" and it leaves me on the JWT token input dialogue box. I have tried restarting home assistant and the Virgin 360 box, but I still get the same "Can't Connect" message.

Thanks for taking the time to read this,

Tom

**Comment by ColinRobbins** [comment]
My guess is you are not correctly obtaining the token from Firefox.

In the Firefox extension, you need to select “open JWT from”, then select “flutter.web_secure_storge…”
See the screenshot in the readme.  https://github.com/Sholofly/lghorizon

You will then see a long token that typically starts “ey….”.  Mine is about 800 characters long.

The mistake others made was using the first token you see in the JWT tool, which is quite sort in comparison.   you need the 800 character “flutter” one.

**Comment by mpsamuels** [comment]
+1 on you've likely got the wrong token. I've literally just set this integration up for a GB based Virgin 360 box using the latest version and made the same mistake at first. 

The first token I saw after the login completed was a fairly short one and I assumed that was what I needed. I got the "Can't Connect" error, went back to check the Firefox extension and saw a much longer token had generated that then worked first time.
